### PR TITLE
Create user crud operations

### PR DESCRIPTION
**Pull request by alex-dishen**

### CODE REVIEW

File: schemas/user.schema.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+export type UserDocument = User & Document<Types.ObjectId>;
+
+export type UserDocumentWithoutPassword = Omit<User, 'password'> &
+  Document<Types.ObjectId>;
+
+@Schema()
+export class User {
+  @Prop({ required: true })
+  first_name: string;
```
Comment (Puchkof) [inline schemas/user.schema.ts]: https://www.mongodb.com/community/forums/t/does-mongodb-have-a-naming-convention-for-collection-and-field-naming/11806/2 I would suggest using camelCase naming convention like in JS/TS

File: schemas/user.schema.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  @Prop({ required: true })
+  last_name: string;
+
+  @Prop({ required: true, select: false })
+  password: string;
```
Comment (Puchkof) [inline schemas/user.schema.ts:18]: Are you sure its a good practice to store password directly in db? It is not secure, you should use password hashing https://docs.nestjs.com/security/encryption-and-hashing

File: src/api/user/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { UserService } from './user.service';
+import { CreateUserDto, UpdateUserDto, UserDto } from './dto/user.dto';
+import { MessageDto } from 'src/shared/dto/message.dto';
+import { PaginatedResult, PaginationDto } from 'src/shared/dto/pagination.dto';
+
+@ApiTags('User')
+@Controller('user')
+export class UserController {
```
Comment (Puchkof) [inline src/api/user/user.controller.ts:19]: 1. Your API is not following REST principles. See https://restfulapi.net/resource-naming/
2. You need some king of authentication(I suggest JWT) to protect endpoints like getting or deleting user. I would suggest endpoints POST /sign-up for registration, POST /sign-in for login, GET /users/current for gettings user info(it should be protected with JWT token and you will get user id from JWT claims), GET /users for getting all user info(it should be available only for admins, so you need roles for users, I would suggest skipping this part for know), PUT /users/current to update current user info(also JWT authentication), DELETE /users/current(also JWT)

File: schemas/user.schema.ts (hunk 2 above)
Comment (alex-dishen) [reply schemas/user.schema.ts:18]: This is just how I named this field. I hash the password on creation and update
Screenshot 2024-08-18 at 18 11 14

File: schemas/user.schema.ts (hunk 1 above)
Comment (alex-dishen) [reply schemas/user.schema.ts]: Just to be clear, when using PostgreSQL the convention is to use snake case and with Mongo camel case, right?

File: src/api/user/user.controller.ts (hunk 3 above)
Comment (alex-dishen) [reply src/api/user/user.controller.ts:19]: Here I just wanted to created simple CRUD operations to understand how to work with Mongo. And I was planning on doing all of the provided suggestions in the next PR. Separate task for authentication

File: schemas/user.schema.ts (hunk 1 above)
Comment (Puchkof) [reply schemas/user.schema.ts]: Actually, there is no strong convention for any of these dbs. With C#, for example, we are using PascalCase. I would suggest to use convenient case for your programming language

File: schemas/user.schema.ts (hunk 2 above)
Comment (Puchkof) [reply schemas/user.schema.ts:18]: Sorry, missed this. Actually, maybes its better to move hashing logic to db layer? If you use repository in another place, your password will be stored without hashing

Comment (alex-dishen) [reply schemas/user.schema.ts:18]: That's a great idea

**Pull request merged**